src/mem/Pool.h

```diff
@@ -171,73 +172,8 @@ class MemPools
     bool defaultIsChunked = false;
 };
 
-/**
- \ingroup MemPoolsAPI
- * a pool is a [growing] space for objects of the same size
- */
-class MemAllocator
-{
-public:
-    MemAllocator (char const *aLabel);
-    virtual ~MemAllocator() {}
-
-    /**
-     \param stats   Object to be filled with statistical data about pool.
-     \retval        Number of objects in use, ie. allocated.
-     */
-    virtual int getStats(MemPoolStats * stats, int accumulate = 0) = 0;
-
-    virtual MemPoolMeter const &getMeter() const = 0;
-
-    /**
-     * Allocate one element from the pool
-     */
-    virtual void *alloc() = 0;
-
-    /**
-     * Free a element allocated by MemAllocator::alloc()
-     */
-    virtual void freeOne(void *) = 0;
-
-    virtual char const *objectType() const;
-    virtual size_t objectSize() const = 0;
-    virtual int getInUseCount() = 0;
-    void zeroBlocks(bool doIt) {doZero = doIt;}
-    int inUseCount();
-
-    /**
-     * Allows you tune chunk size of pooling. Objects are allocated in chunks
-     * instead of individually. This conserves memory, reduces fragmentation.
-     * Because of that memory can be freed also only in chunks. Therefore
-     * there is tradeoff between memory conservation due to chunking and free
-     * memory fragmentation.
-     *
-     \note  As a general guideline, increase chunk size only for pools that keep
-     *      very many items for relatively long time.
-     */
-    virtual void setChunkSize(size_t) {}
-
-    /**
-     \param minSize Minimum size needed to be allocated.
-     \retval n Smallest size divisible by sizeof(void*)
-     */
-    static size_t RoundedSize(size_t minSize);
-
-protected:
-    /** Whether to zero memory on initial allocation and on return to the pool.
-     *
-     * We do this on some pools because many object constructors are/were incomplete
-     * and we are afraid some code may use the object after free.
-     * These probems are becoming less common, so when possible set this to false.
-     */
-    bool doZero;
-
-private:
-    const char *label;
-};
-
 /// \ingroup MemPoolsAPI
-class MemImplementingAllocator : public MemAllocator
+class MemImplementingAllocator : public Mem::Allocator
 {
 public:
     MemImplementingAllocator(char const *aLabel, size_t aSize);
```

Comment:
> PR description: Update and add missing documentation for class members. Including some child classes.

Please remove overriden method _descriptions_ below since we are going to be polishing their original/true descriptions in Mem::Allocator. Please check grandkids as well.

Reply:
Done in[b48d53b](https://github.com/squid-cache/squid/pull/1181/commits/b48d53b23db6f2b01f52b8b94f7a63e7e6a31ea9)